Feat post page

anangyoga opened the pull request:
**Changes/Features:**
I created the Post page based on the UI Design. I also created a small routing where we can see the Post page.

**Testing:**
1. Upvotes Component
- I tested for component renders
- I tested for both buttons render
- I tested to display correct props count

2. Homepage
- I tested for Homepage renders
- I tested for Homepage to have 3 columns

3. Post
- I tested for Post renders
- I tested for Post has hero image
- I tested for Post has 6 images
- I tested for users can write comment

anangyoga commented:
hi kak @fatjan this PR is ready for Review.
I've made the test and all passed.

![unit-test](https://github.com/project-zusammen/hidden-gem-frontend/assets/67864789/fae150c4-9d59-48b3-b5d4-7d8d49308f22)

fatjan commented on src/data/chipLists.ts:
```diff
@@ -0,0 +1,24 @@
+const chipLists = [
```
Is there a specific reason of making this file a ts one not js?

also the other file in the same dir

fatjan commented on src/pages/Homepage.jsx:
```diff
@@ -0,0 +1,28 @@
+import { Container, Grid } from "@mui/material";
```
please also add a test file for this file

anangyoga replied:
I have tested this file.

fatjan commented on src/pages/Post.jsx:
```diff
@@ -0,0 +1,132 @@
+import { ImageList, ImageListItem, Box, Card, CardMedia, Container, Grid, Typography, Chip, IconButton, Paper, Stack, TextField, Button, Avatar } from "@mui/material";
```
please also add a test file for this file

anangyoga replied:
I also have tested this.

fatjan commented:
> _ No description provided. _

Please add some description in your next PRs, thanks

anangyoga reviewed (commented):
hi kak, I have made some changes based on your suggestion. let me know if there's more I can do. thank you.

fatjan commented:
> **Changes/Features:** I created the Post page based on the UI Design. I also created a small routing where we can see the Post page.
> 
> **Testing:**
> 
> 1. Upvotes Component
> 
> * I tested for component renders
> * I tested for both buttons render
> * I tested to display correct props count
> 
> 2. Homepage
> 
> * I tested for Homepage renders
> * I tested for Homepage to have 3 columns
> 
> 3. Post
> 
> * I tested for Post renders
> * I tested for Post has hero image
> * I tested for Post has 6 images
> * I tested for users can write comment

Great description, would be better to include the ss of what you did as well. Let's do it on the next PR.

Example:
![Screenshot 2024-01-25 at 16 51 04](https://github.com/project-zusammen/hidden-gem-frontend/assets/46950501/b989877c-0e6f-4033-840f-793e1e1b2b54)

Is this correct or am I missing something here?

fatjan commented:
> > **Changes/Features:** I created the Post page based on the UI Design. I also created a small routing where we can see the Post page.
> > **Testing:**
> > 
> > 1. Upvotes Component
> > 
> > 
> > * I tested for component renders
> > * I tested for both buttons render
> > * I tested to display correct props count
> > 
> > 
> > 2. Homepage
> > 
> > 
> > * I tested for Homepage renders
> > * I tested for Homepage to have 3 columns
> > 
> > 
> > 3. Post
> > 
> > 
> > * I tested for Post renders
> > * I tested for Post has hero image
> > * I tested for Post has 6 images
> > * I tested for users can write comment
> 
> Great description, would be better to include the ss of what you did as well. Let's do it on the next PR.
> 
> Example: ![Screenshot 2024-01-25 at 16 51 04](https://private-user-images.githubusercontent.com/46950501/299615034-b989877c-0e6f-4033-840f-793e1e1b2b54.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxNzY1ODcsIm5iZiI6MTcwNjE3NjI4NywicGF0aCI6Ii80Njk1MDUwMS8yOTk2MTUwMzQtYjk4OTg3N2MtMGU2Zi00MDMzLTg0MGYtNzkzZTFlMWIyYjU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDA5NTEyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzMjJjYTQwODBmNDA2NDE5M2FmZWNjNzBkZDQ2Y2I2M2VmOWRlYzA1MTFjMjc4YjkyYmIwOGFkNzNkZWU0MTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uSSWrsy1z5DmMpPheUkqe5LG_gp8N65Ooo2VcGoMP68)
> 
> Is this correct or am I missing something here?

This is what I see when running the app using your branch

fatjan commented:
> > > **Changes/Features:** I created the Post page based on the UI Design. I also created a small routing where we can see the Post page.
> > > **Testing:**
> > > 
> > > 1. Upvotes Component
> > > 
> > > 
> > > * I tested for component renders
> > > * I tested for both buttons render
> > > * I tested to display correct props count
> > > 
> > > 
> > > 2. Homepage
> > > 
> > > 
> > > * I tested for Homepage renders
> > > * I tested for Homepage to have 3 columns
> > > 
> > > 
> > > 3. Post
> > > 
> > > 
> > > * I tested for Post renders
> > > * I tested for Post has hero image
> > > * I tested for Post has 6 images
> > > * I tested for users can write comment
> > 
> > 
> > Great description, would be better to include the ss of what you did as well. Let's do it on the next PR.
> > Example: ![Screenshot 2024-01-25 at 16 51 04](https://private-user-images.githubusercontent.com/46950501/299615034-b989877c-0e6f-4033-840f-793e1e1b2b54.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxNzY1ODcsIm5iZiI6MTcwNjE3NjI4NywicGF0aCI6Ii80Njk1MDUwMS8yOTk2MTUwMzQtYjk4OTg3N2MtMGU2Zi00MDMzLTg0MGYtNzkzZTFlMWIyYjU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDA5NTEyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzMjJjYTQwODBmNDA2NDE5M2FmZWNjNzBkZDQ2Y2I2M2VmOWRlYzA1MTFjMjc4YjkyYmIwOGFkNzNkZWU0MTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uSSWrsy1z5DmMpPheUkqe5LG_gp8N65Ooo2VcGoMP68)
> > Is this correct or am I missing something here?
> 
> This is what I see when running the app using your branch

Please rename the page to Review instead of Post

And here's what it looks like:
I think it would be better if we add some spaces on the left and right, we can discuss more on this during tonight's meeting
![Screenshot 2024-01-25 at 17 39 37](https://github.com/project-zusammen/hidden-gem-frontend/assets/46950501/3dd935f6-2eb3-4aed-af57-86d78ec109f9)

The figma:
![Screenshot 2024-01-25 at 17 38 12](https://github.com/project-zusammen/hidden-gem-frontend/assets/46950501/975fd0ce-6a45-4ba5-96a0-25db63020b86)

anangyoga commented:
hi kak @fatjan there are some changes I did:
- I have changed the /post path to /review.

- I have added temporary 'button' on the homepage to navigate to /review page.
![review](https://github.com/project-zusammen/hidden-gem-frontend/assets/67864789/fadccf89-5520-4a6d-a5fd-28244012c050)


- I also have adjusted the left-right padding to match the Footer's padding. I can increase or decrease it now but so far, I make it has same padding with Footer component.
![padding](https://github.com/project-zusammen/hidden-gem-frontend/assets/67864789/4e00c482-dc50-42b9-86dd-61414d0e35d1)

I'm open for any revision. Thank you.
